mmdeploy/codebase/mmdet/deploy/object_detection.py

```diff
@@ -214,24 +214,58 @@ def get_preprocess(self, *args, **kwargs) -> Dict:
         """
         input_shape = get_input_shape(self.deploy_cfg)
         model_cfg = process_model_config(self.model_cfg, [''], input_shape)
-        preprocess = model_cfg.test_pipeline
-        return preprocess
+        pipeline = model_cfg.test_pipeline
+        meta_keys = [
+            'filename', 'ori_filename', 'ori_shape', 'img_shape', 'pad_shape',
+            'scale_factor', 'flip', 'flip_direction', 'img_norm_cfg',
+            'valid_ratio'
+        ]
+        transforms = [
+            item for item in pipeline if 'Random' not in item['type']
+            and 'Annotation' not in item['type']
+        ]
+        for i, transform in enumerate(transforms):
+            if transform['type'] == 'PackDetInputs':
+                meta_keys += transform[
+                    'meta_keys'] if 'meta_keys' in transform else []
+                transform['meta_keys'] = list(set(meta_keys))
+                transform['keys'] = ['img']
+                transforms[i]['type'] = 'Collect'
+            if transform['type'] == 'Resize':
+                transforms[i]['size'] = transforms[i]['scale']
+
+        data_preprocessor = model_cfg.model.data_preprocessor
+        transforms.insert(-1, dict(type='DefaultFormatBundle'))
+        transforms.insert(
+            -2,
+            dict(
+                type='Pad',
+                size_divisor=data_preprocessor.get('pad_size_divisor', 1)))
+        transforms.insert(
+            -3,
+            dict(
+                type='Normalize',
+                to_rgb=data_preprocessor.get('bgr_to_rgb', False),
```

Comment:
Isn't the default value of `to_rgb` `True`?

Reply:
The default value of it in `ImgDataPreprocessor` is False.